### PR TITLE
Refactor DigitalCovidCertificate JSON/UI interfaces (DEV)

### PR DESCRIPTION
**Pull request by d4rken**
Rename and move classes/interfaces related to qrcode (json) and interfaces used for our UIs.

Next features will require us to group multiple certificates, if they share common interfaces, it will be easier to work with.

Merge after #3381

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/certificate/Dcc.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            firstNameStandardized = nameData.givenNameStandardized
+        )
+
+    interface Payload {
+        val targetId: String
+        val certificateCountry: String
+        val certificateIssuer: String
+        val uniqueCertificateIdentifier: String
+    }
+}
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/certificate/Dcc.kt:50]: Common interface for properties shared by all DCC qrcodes.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/certificate/DccData.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package de.rki.coronawarnapp.covidcertificate.common.certificate
+
+data class DccData<CertT : Dcc<*>>(
+    val header: DccHeader,
+    val certificate: CertT,
+)
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/certificate/DccData.kt:6]: Common representation for a scanned qrcode.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/qrcode/DccQrCode.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+interface DccQrCode<DccT : Dcc<*>> {
+    val qrCode: QrCodeString
+    val data: DccData<DccT>
+
+    val personIdentifier: CertificatePersonIdentifier
+        get() = data.certificate.personIdentifier
+
+    val uniqueCertificateIdentifier: String
+        get() = data.certificate.payload.uniqueCertificateIdentifier
+}
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/qrcode/DccQrCode.kt:16]: Common result for a qrcode extractor.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/certificate/CwaCovidCertificate.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    val lastName: String
+    val fullName: String
+    val dateOfBirth: LocalDate
+
+    val personIdentifier: CertificatePersonIdentifier
+
+    val certificateIssuer: String
+    val certificateCountry: String
+    val certificateId: String
+}
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/certificate/CwaCovidCertificate.kt:29]: Common interface for user facing certificate data (`VaccinationCertificate`, `TestCertificate`),

**Review by chiljamgossow — APPROVED** [review]
👌

**Review by BMItr — APPROVED** [review]
lgtm

**Pull request merged**